test-workloads/src/bin/tellus.rs

```diff
@@ -81,6 +91,87 @@ fn set_vcpu_reg(vmid: u64, register: sbi::TvmCpuRegister, value: u64) {
     tsm::set_vcpu_reg(vmid, 0, register, value).expect("Tellus - TvmCpuGetRegister failed")
 }
 
+fn exercise_pmu_functionality() {
+    fn generate_counter_mask(high: u64) -> u64 {
+        (1 << high) - 1
+    }
+
+    /// Safety: Caller must ensure that that CSR exists on the platform, and that the read doesn't have unexpected side-effects
```

Comment:
Could you just pull in the `riscv-regs` crate and use that?

Reply:
Changed.